com.unity.shadergraph/Editor/GraphUI/ShaderGraphMainToolbar.cs

```diff
@@ -1,11 +1,135 @@
+        {
+            // Get rid of all existing visual elements defined by MainToolbar and add them in order we need here
+            Clear();
+
+            AddToClassList(ussClassName);
+
+            this.AddStylesheet("ShaderGraphMainToolbar.uss");
+            this.AddStylesheet(EditorGUIUtility.isProSkin ? "ShaderGraphMainToolbar_dark.uss" : "ShaderGraphMainToolbar_light.uss");
+
+            var tpl = GraphElementHelper.LoadUXML("ShaderGraphMainToolbar.uxml");
+            tpl.CloneTree(this);
+
+            m_SaveButton = this.MandatoryQ<ToolbarButton>(SaveButton);
+            m_SaveButton.ChangeClickEvent(OnSaveButton);
+            Add(m_SaveButton);
+
+            m_SaveAsButton = this.MandatoryQ<ToolbarButton>(SaveAsButton);
+            m_SaveAsButton.ChangeClickEvent(OnSaveAsButton);
+            Add(m_SaveAsButton);
+
+            m_ShowInProjectButton = this.MandatoryQ<ToolbarButton>(ShowInProjectButton);
+            m_ShowInProjectButton.ChangeClickEvent(OnShowInProjectButton);
+            Add(m_ShowInProjectButton);
+
+            //m_CheckOutButton = this.MandatoryQ<ToolbarButton>(CheckOutButton);
+            //m_CheckOutButton.ChangeClickEvent(OnCheckOutButton);
+            //Add(m_CheckOutButton);
+
+            Add(new ToolbarSpacer());
+
+            Add(m_Breadcrumb);
+
+            var flexibleSpacer = this.MandatoryQ<ToolbarSpacer>(Spacer);
+            Add(flexibleSpacer);
+
+            //Add(new Label("Color Mode"));
+
+            //m_ColorModeDropdown = this.MandatoryQ<EnumField>(ColorModeDropdown);
+            //m_ColorModeDropdown.RegisterCallback<ChangeEvent<Enum>>(OnColorModeChanged);
+            //m_ColorModeDropdown.Init(ColorMode.None);
+            //Add(m_ColorModeDropdown);
+            //
+            //m_ToggleBlackboardButton = this.MandatoryQ<ToolbarButton>(ToggleBlackboardButton);
+            //m_ToggleBlackboardButton.ChangeClickEvent(OnToggleBlackboardButton);
+            //Add(m_ToggleBlackboardButton);
+            //
+            //m_ToggleInspectorButton = this.MandatoryQ<ToolbarButton>(ToggleGraphInspectorButton);
+            //m_ToggleInspectorButton.ChangeClickEvent(OnToggleInspectorButton);
+            //Add(m_ToggleInspectorButton);
+            //
+            //m_ToggleMainPreviewButton = this.MandatoryQ<ToolbarButton>(ToggleMainPreviewButton);
+            //m_ToggleMainPreviewButton.ChangeClickEvent(OnTogglePreviewButton);
+            //Add(m_ToggleMainPreviewButton);
+
+            Add(m_OptionsButton);
+        }
+
+        void OnSaveButton()
+        {
+        }
+
+        void OnSaveAsButton()
+        {
+        }
+
+        void OnShowInProjectButton()
+        {
+            var path = GraphTool.ToolState.CurrentGraph.GetGraphAssetModelPath();
+            var asset = AssetDatabase.LoadAssetAtPath<UnityEngine.Object>(path);
+            EditorGUIUtility.PingObject(asset);
+        }
+
+        void OnCheckOutButton()
+        {
```

Comment:
Could you please have these emit "UNIMPLEMENTED" warnings into the console and add TODO comments here so that we find these before release if they are unimplemented?

Reply:
Can do!

Reply:
Ty